core/arch/emit_utils_shared.c

```diff
@@ -4923,10 +4934,12 @@ emit_fcache_enter_gonative(dcontext_t *dcontext, generated_code_t *code, byte *p
     APP(&ilist,
         XINST_CREATE_store(dcontext, OPND_CREATE_MEMPTR(DR_REG_SP, -XSP_SZ),
                            opnd_create_reg(DR_REG_R0)));
-    /* get target PC */
+    /* Load target PC from FCACHE_ENTER_TARGET_SLOT, stored by
+     * by append_setup_fcache_target.
+     */
     APP(&ilist,
         XINST_CREATE_load(dcontext, opnd_create_reg(DR_REG_R0),
-                          OPND_CREATE_MEMPTR(dr_reg_stolen, 16)));
+                          OPND_CREATE_MEMPTR(dr_reg_stolen, FCACHE_ENTER_TARGET_SLOT)));
```

Comment:
This could be instr_create_restore_from_tls()?

Reply:
Done.